Dockerfile

```diff
@@ -0,0 +1,33 @@
+#
+# Stage: Development build
+#
+FROM node:lts-alpine as dev
+
+WORKDIR /app
+
+COPY  ./package.json \
+      ./package-lock.json \
+      ./tsconfig.json \
+      ./
+
+RUN npm install
+
+COPY src/ src/
+
+#
+# Stage: Production build
+#
+FROM dev as build-prod
+
+COPY --from=dev /app/ .
+RUN npm run build
+
+#
+# Stage: Production
+#
+FROM node:lts-alpine as prod
```

Comment:
Future optimisation, although depends on contents of `package.json`. Post build you won't need any dev dependencies that were installed for the build, so you could have a second stage to `npm install --only=prod` and that could reduce the container size a fair bit.
Not worth resolving now though, but something to consider in the future.

Reply:
I had messed around with `npm prune --production` but the resulting docker image still turned out to be the same size. I think theres more to it but e should look into it in future.